### ISSUE TITLE
Support LaTeX ?

### ISSUE THREAD

**Issue by phucynwa** [halilozercan/compose-richtext#70]
I've just found this awesome library. I love your attempt.

I hope that compose-richtext will support LaTeX soon.

**Comment by halilozercan** [comment]
Rendering LaTeX would be an almost impossible task for any Android library alone. I don't think we will have that kind of ambitious goal anytime soon. If we are talking about a small subset of LaTeX like math equations, that'd be something we can look into. I mean not the parsing of a math syntax but providing necessary builders to write a generic math function.

```
buildFormula {
  pow(a, 2) + pow(b, 2) eq pow(c,2)
}
```

would render something like

![image](https://user-images.githubusercontent.com/5561132/145686584-71759974-f51e-425c-9fb8-19301cd0e7fa.png)

**Comment by phucynwa** [comment]
@halilozercan Yes. I wish `richtext-commonmark` will support LaTeX syntax like Markwon library.
I hope I can use this function soon. Thank you.

**Comment by leon2017** [comment]
请问现在支持LaTex了吗

**Comment by zayass** [comment]
I am not sure about quality but here some existing libraries for parsing/rendering LaTeX on android

https://github.com/NanoMichael/AndroidLaTeXMath
https://github.com/noties/jlatexmath-android

@halilozercan what do you think about some kind of plugins to parse/render embedded things.

I think it is better than inventing yet another math markup language. Because it is a pretty common and widely used scenario to embed LaTeX into markdown:


```latex
f{\left( a \right )} = \frac{1}{2 \pi i} \oint_{\gamma} \frac{f{\left( z \right )}}{z - a} dz
```

$$f{\left( a \right )} = \frac{1}{2 \pi i} \oint_{\gamma} \frac{f{\left( z \right )}}{z - a} dz$$

**Comment by sakibanda** [comment]
Hi, Can anyone please update on this? I need Latex support into markdown.

**Comment by phucynwa** [comment]
@sakibanda I spent a night making it possible 😄 
```
dependencyResolutionManagement {
    ...
    repositories {
        ...
        maven("https://jitpack.io") <-- add this line
    }
}
```
```
implementation("com.github.phucynwa.compose-richtext:richtext-commonmark:support_latex-SNAPSHOT")
implementation("com.github.phucynwa.compose-richtext:richtext-ui:support_latex-SNAPSHOT")
implementation("com.github.phucynwa.compose-richtext:richtext-ui-material3:support_latex-SNAPSHOT")
```

You can check my branch and modify it as you want. https://github.com/phucynwa/compose-richtext/tree/support_latex

~~I copied the compose code from https://github.com/yicu23/LaTeX.kt~~
I used https://github.com/noties/jlatexmath-android for rendering LaTeX since it support more syntaxes.

**Comment by sakibanda** [comment]
Sorry couldn't reply earlier.  Thanks a lot @phucynwa, I appreciate it 👍

I will implement and let you know how it goes.

**Comment by sakibanda** [comment]
When i switch to your branch & try android build without making any changes I get these errors:

Unresolved reference: com.halilibo.richtext.ui.RichTextThemeIntegration
Unresolved reference: com.halilibo.richtext.ui.material3.Material3RichText
Unresolved reference: com.halilibo.richtext.markdown.Markdown

**Comment by sakibanda** [comment]
Never mind. followed the documention here : 
https://halilibo.com/compose-richtext/richtext-ui-material3/

Then, build worked but Latex functions are not rendering, I am trying everything but don't know why its not working, please find below code & screenshot.

Here is my code:
```
RichText(
    modifier = Modifier.padding(horizontal = 0.dp, vertical = 0.dp),
    style = RichTextStyle( codeBlockStyle = CodeBlockStyle()) {
    Markdown(
        content = message.content.trimIndent()
            .replace("\n- ", "\n    - ")
    )
}
```
![Screenshot_2024-12-22-11-31-53-945_app amazeai android](https://github.com/user-attachments/assets/a8a97d78-2014-4b0c-bd1e-eba18a3c75c2) 


Please help.

**Comment by sakibanda** [comment]
Hello,

Any update on this? Tried the code from your sample, but still its not rendering LaTex code.

Please help.

**Comment by darriousliu** [comment]
I created a library [KotlinTex](https://github.com/darriousliu/KotlinTeX) for rendering latex in Compose Multiplatform by using freetype, you can have a try.

**Comment by zly2006** [comment]
I found a new lib for this, https://github.com/huarangmeng/latex. And I am trying to integrate it into my own project, I have made it working:

<img width="812" height="874" alt="Image" src="https://github.com/user-attachments/assets/bbee7cf1-8427-4df1-9347-cd04815d1a88" />
